File: src/main/java/org/interview/twitter/model/Author.java
package org.interview.twitter.model;

import javax.persistence.*;
import java.util.Date;
import java.util.Set;

@Entity
@Table(name = "Author")
public class Author {

    @Id
    @Column(name = "USER_ID")
    private Long userId;

    @Column(name = "CREATION_DATE")
    private Date creationDate;

    @Column(name = "AUTHOR_NAME")
    private String name;

    @Column(name = "SCREEN_NAME")
    private String screenName;

    @ManyToMany(mappedBy = "authors")
    private Set<TwitterMessage> twitterMessages;

    public Long getUserId() {
        return userId;
    }

    public Date getCreationDate() {
        return creationDate;
    }

    public String getName() {
        return name;
    }

    public String getScreenName() {
        return screenName;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public void setCreationDate(Date creationDate) {
        this.creationDate = creationDate;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setScreenName(String screenName) {
        this.screenName = screenName;
    }

    public Set<TwitterMessage> getTwitterMessages() {
        return twitterMessages;
    }

    public void setTwitterMessages(Set<TwitterMessage> twitterMessages) {
        this.twitterMessages = twitterMessages;
    }
}

File: src/main/kotlin/org/interview/twitter/model/Author.kt
package org.interview.twitter.model

import javax.persistence.*
import java.util.Date

/**
 * @authors Mehdi Najafian
 */
@Entity
@Table(name = "Author")
class Author : Comparable<Author> {
    //this constructor is for Jpa
    constructor()

    constructor(userId: Long?, creationDate: Date?, name: String?, screenName: String?, twitterMessage: List<TwitterMessage>?) {
        this.userId = userId
        this.creationDate = creationDate
        this.name = name
        this.screenName = screenName
        this.twitterMessage = twitterMessage
    }

    @Id
    @Column(name = "USER_ID",unique = true)
    var userId: Long? = null

    @Column(name = "CREATION_DATE")
    var creationDate: Date? = null

    @Column(name = "AUTHOR_NAME")
    var name: String? = null

    @Column(name = "SCREEN_NAME")
    var screenName: String? = null

    @OneToMany(mappedBy = "author")
    var twitterMessage: List<TwitterMessage>? = null

    override fun compareTo(other: Author): Int {
        var result = 0
        if (other !== this)
            result = this.creationDate!!.compareTo(other.creationDate)
        return result
    }

}
